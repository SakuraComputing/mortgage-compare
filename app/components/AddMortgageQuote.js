import React from 'react';

class NewMortgageQuote extends React.Component {
  constructor(props) {
    super(props);
    this.state = { lender: 'Santander'};

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange() {
    this.setState({value: event.target.value});
  }

  handleSubmit() {
    event.preventDefault();
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="">
          New Mortgage Lender
            <select value={this.state.lender} onChange={this.handleChange}>
              <option value="Halifax" icon="/assets/images/Halifax.png"></option>
              <option value="Santander" icon="/assets/images/santander.png"></option>
            </select>
            <button>Cancel</button>
        </label>
      </form>
    )
  }
};

export default NewMortgageQuote;
