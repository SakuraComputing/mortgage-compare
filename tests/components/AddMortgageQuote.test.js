import React from 'react';
import { shallow } from 'enzyme';
import AddMortgageQuote from '../../app/components/AddMortgageQuote';

describe('New Quote',() => {
  it('should render the form correctly', () => {
    const wrapper = shallow(<AddMortgageQuote />);
    expect(wrapper).toMatchSnapshot();
  });

});
