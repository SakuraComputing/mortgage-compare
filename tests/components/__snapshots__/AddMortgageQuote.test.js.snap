// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Quote should render the form correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewMortgageQuote />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <label
        htmlFor=""
      >
        New Mortgage Lender
        <select
          onChange={[Function]}
          value="Santander"
        >
          <option
            icon="/assets/images/Halifax.png"
            value="Halifax"
          />
          <option
            icon="/assets/images/santander.png"
            value="Santander"
          />
        </select>
        <button>
          Cancel
        </button>
      </label>,
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "New Mortgage Lender",
          <select
            onChange={[Function]}
            value="Santander"
          >
            <option
              icon="/assets/images/Halifax.png"
              value="Halifax"
            />
            <option
              icon="/assets/images/santander.png"
              value="Santander"
            />
          </select>,
          <button>
            Cancel
          </button>,
        ],
        "htmlFor": "",
      },
      "ref": null,
      "rendered": Array [
        "New Mortgage Lender",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                icon="/assets/images/Halifax.png"
                value="Halifax"
              />,
              <option
                icon="/assets/images/santander.png"
                value="Santander"
              />,
            ],
            "onChange": [Function],
            "value": "Santander",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "icon": "/assets/images/Halifax.png",
                "value": "Halifax",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "icon": "/assets/images/santander.png",
                "value": "Santander",
              },
              "ref": null,
              "rendered": null,
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
      ],
      "type": "label",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <label
          htmlFor=""
        >
          New Mortgage Lender
          <select
            onChange={[Function]}
            value="Santander"
          >
            <option
              icon="/assets/images/Halifax.png"
              value="Halifax"
            />
            <option
              icon="/assets/images/santander.png"
              value="Santander"
            />
          </select>
          <button>
            Cancel
          </button>
        </label>,
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "New Mortgage Lender",
            <select
              onChange={[Function]}
              value="Santander"
            >
              <option
                icon="/assets/images/Halifax.png"
                value="Halifax"
              />
              <option
                icon="/assets/images/santander.png"
                value="Santander"
              />
            </select>,
            <button>
              Cancel
            </button>,
          ],
          "htmlFor": "",
        },
        "ref": null,
        "rendered": Array [
          "New Mortgage Lender",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  icon="/assets/images/Halifax.png"
                  value="Halifax"
                />,
                <option
                  icon="/assets/images/santander.png"
                  value="Santander"
                />,
              ],
              "onChange": [Function],
              "value": "Santander",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "icon": "/assets/images/Halifax.png",
                  "value": "Halifax",
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "icon": "/assets/images/santander.png",
                  "value": "Santander",
                },
                "ref": null,
                "rendered": null,
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cancel",
            },
            "ref": null,
            "rendered": "Cancel",
            "type": "button",
          },
        ],
        "type": "label",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
